Use a BitSet to hold quirk state

brentfulgham opened the pull request:
#### 45c24f4e4c763cb4cf586f1f87b685c485743119
<pre>
Use a BitSet to hold quirk state
<a href="https://bugs.webkit.org/show_bug.cgi?id=302333">https://bugs.webkit.org/show_bug.cgi?id=302333</a>
<a href="https://rdar.apple.com/164490282">rdar://164490282</a>

Reviewed by Brandon Stewart.

Rather than an ever-growing set of `bool` values, let&apos;s represent quirks state using a
BitSet. This allows us to do a quick check of whether *any* quirks are active in the
current domain, rather than individually testing each one. In theory, this should be a
small performance benefit since most sites do not have any relevant quirk and quickly
determining this requires fewer steps.

Tested by existing Quirks regression tests.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::needsFormControlToBeMouseFocusable const):
(WebCore::Quirks::needsAutoplayPlayPauseEvents const):
(WebCore::Quirks::needsSeekingSupportDisabled const):
(WebCore::Quirks::needsPerDocumentAutoplayBehavior const):
(WebCore::Quirks::shouldAutoplayWebAudioForArbitraryUserGesture const):
(WebCore::Quirks::hasBrokenEncryptedMediaAPISupportQuirk const):
(WebCore::Quirks::isTouchBarUpdateSuppressedForHiddenContentEditable const):
(WebCore::Quirks::isNeverRichlyEditableForTouchBar const):
(WebCore::Quirks::shouldSuppressAutocorrectionAndAutocapitalizationInHiddenEditableAreas const):
(WebCore::Quirks::shouldDispatchSyntheticMouseEventsWhenModifyingSelection const):
(WebCore::Quirks::needsYouTubeMouseOutQuirk const):
(WebCore::Quirks::shouldDisableWritingSuggestionsByDefault const):
(WebCore::Quirks::storageAccessUserAgentStringQuirkForDomain):
(WebCore::Quirks::shouldDisableElementFullscreenQuirk const):
(WebCore::Quirks::shouldDispatchSimulatedMouseEvents const):
(WebCore::Quirks::shouldDispatchedSimulatedMouseEventsAssumeDefaultPrevented const):
(WebCore::Quirks::shouldPreventDispatchOfTouchEvent const):
(WebCore::Quirks::shouldAvoidResizingWhenInputViewBoundsChange const):
(WebCore::Quirks::shouldDisablePointerEventsQuirk const):
(WebCore::Quirks::needsDeferKeyDownAndKeyPressTimersUntilNextEditingCommand const):
(WebCore::Quirks::needsGMailOverflowScrollQuirk const):
(WebCore::Quirks::needsYouTubeOverflowScrollQuirk const):
(WebCore::Quirks::needsPrimeVideoUserSelectNoneQuirk const):
(WebCore::Quirks::needsFacebookRemoveNotSupportedQuirk const):
(WebCore::Quirks::needsScrollbarWidthThinDisabledQuirk const):
(WebCore::Quirks::needsBodyScrollbarWidthNoneDisabledQuirk const):
(WebCore::Quirks::needsFullscreenDisplayNoneQuirk const):
(WebCore::Quirks::needsFullscreenObjectFitQuirk const):
(WebCore::Quirks::needsZomatoEmailLoginLabelQuirk const):
(WebCore::Quirks::needsGoogleMapsScrollingQuirk const):
(WebCore::Quirks::needsGoogleTranslateScrollingQuirk const):
(WebCore::Quirks::shouldSilenceResizeObservers const):
(WebCore::Quirks::shouldSilenceWindowResizeEventsDuringApplicationSnapshotting const):
(WebCore::Quirks::shouldSilenceMediaQueryListChangeEvents const):
(WebCore::Quirks::shouldAvoidScrollingWhenFocusedContentIsVisible const):
(WebCore::Quirks::shouldUseLegacySelectPopoverDismissalBehaviorInDataActivation const):
(WebCore::Quirks::shouldIgnoreAriaForFastPathContentObservationCheck const):
(WebCore::Quirks::shouldIgnoreViewportArgumentsToAvoidExcessiveZoom const):
(WebCore::Quirks::shouldIgnoreViewportArgumentsToAvoidEnlargedView const):
(WebCore::Quirks::shouldOpenAsAboutBlank const):
(WebCore::Quirks::needsPreloadAutoQuirk const):
(WebCore::Quirks::shouldBypassBackForwardCache const):
(WebCore::Quirks::shouldBypassAsyncScriptDeferring const):
(WebCore::Quirks::shouldEnableFacebookFlagQuirk const):
(WebCore::Quirks::applyFacebookFlagQuirk):
(WebCore::Quirks::shouldEnableLegacyGetUserMediaQuirk const):
(WebCore::Quirks::shouldDisableImageCaptureQuirk const):
(WebCore::Quirks::shouldEnableCameraAndMicrophonePermissionStateQuirk const):
(WebCore::Quirks::shouldEnableRemoteTrackLabelQuirk const):
(WebCore::Quirks::shouldEnableSpeakerSelectionPermissionsPolicyQuirk const):
(WebCore::Quirks::shouldEnableEnumerateDeviceQuirk const):
(WebCore::Quirks::shouldEnableRTCEncodedStreamsQuirk const):
(WebCore::Quirks::shouldUnloadHeavyFrame const):
(WebCore::Quirks::needsCanPlayAfterSeekedQuirk const):
(WebCore::Quirks::shouldLayOutAtMinimumWindowWidthWhenIgnoringScalingConstraints const):
(WebCore::Quirks::shouldNotAutoUpgradeToHTTPSNavigation):
(WebCore::Quirks::triggerOptionalStorageAccessQuirk const):
(WebCore::Quirks::needsVP9FullRangeFlagQuirk const):
(WebCore::Quirks::requiresUserGestureToPauseInPictureInPicture const):
(WebCore::Quirks::returnNullPictureInPictureElementDuringFullscreenChange const):
(WebCore::Quirks::requiresUserGestureToLoadInPictureInPicture const):
(WebCore::Quirks::blocksReturnToFullscreenFromPictureInPictureQuirk const):
(WebCore::Quirks::blocksEnteringStandardFullscreenFromPictureInPictureQuirk const):
(WebCore::Quirks::shouldDisableEndFullscreenEventWhenEnteringPictureInPictureFromFullscreenQuirk const):
(WebCore::Quirks::allowLayeredFullscreenVideos const):
(WebCore::Quirks::shouldDisableFullscreenVideoAspectRatioAdaptiveSizing const):
(WebCore::Quirks::shouldEnableFontLoadingAPIQuirk const):
(WebCore::Quirks::shouldDisableAdSkippingInPip const):
(WebCore::Quirks::needsVideoShouldMaintainAspectRatioQuirk const):
(WebCore::Quirks::shouldExposeShowModalDialog const):
(WebCore::Quirks::shouldNavigatorPluginsBeEmpty const):
(WebCore::Quirks::shouldDisableLazyIframeLoadingQuirk const):
(WebCore::Quirks::shouldDisableFetchMetadata const):
(WebCore::Quirks::shouldBlockFetchWithNewlineAndLessThan const):
(WebCore::Quirks::shouldDisablePushStateFilePathRestrictions const):
(WebCore::Quirks::needsConfigurableIndexedPropertiesQuirk const):
(WebCore::Quirks::advancedPrivacyProtectionSubstituteDataURLForScriptWithFeatures const):
(WebCore::Quirks::needsResettingTransitionCancelsRunningTransitionQuirk const):
(WebCore::Quirks::shouldDisableDataURLPaddingValidation const):
(WebCore::Quirks::needsDisableDOMPasteAccessQuirk const):
(WebCore::Quirks::shouldPreventOrientationMediaQueryFromEvaluatingToLandscape const):
(WebCore::Quirks::shouldFlipScreenDimensions const):
(WebCore::Quirks::requirePageVisibilityToPlayAudioQuirk const):
(WebCore::Quirks::shouldIgnorePlaysInlineRequirementQuirk const):
(WebCore::Quirks::shouldUseEphemeralPartitionedStorageForDOMCookies const):
(WebCore::Quirks::needsLaxSameSiteCookieQuirk const):
(WebCore::Quirks::shouldIgnoreTextAutoSizing const):
(WebCore::Quirks::scriptToEvaluateBeforeRunningScriptFromURL):
(WebCore::Quirks::shouldHideCoarsePointerCharacteristics const):
(WebCore::Quirks::implicitMuteWhenVolumeSetToZero const):
(WebCore::Quirks::shouldDispatchPointerOutAfterHandlingSyntheticClick const):
(WebCore::Quirks::needsZeroMaxTouchPointsQuirk const):
(WebCore::Quirks::needsChromeMediaControlsPseudoElement const):
(WebCore::Quirks::shouldHideSoftTopScrollEdgeEffectDuringFocus const):
(WebCore::Quirks::shouldTreatAddingMouseOutEventListenerAsContentChange const):
(WebCore::Quirks::shouldSynthesizeTouchEventsAfterNonSyntheticClick const):
(WebCore::Quirks::shouldIgnoreContentObservationForClick const):
(WebCore::Quirks::needsMozillaFileTypeForDataTransfer const):
(WebCore::Quirks::needsBingGestureEventQuirk const):
(WebCore::Quirks::shouldAvoidStartingSelectionOnMouseDownOverPointerCursor const):
(WebCore::Quirks::shouldReuseLiveRangeForSelectionUpdate const):
(WebCore::Quirks::needsPointerTouchCompatibility const):
(WebCore::Quirks::needsFacebookStoriesCreationFormQuirk const):
(WebCore::Quirks::needsHotelsAnimationQuirk const):
(WebCore::Quirks::needsClaudeSidebarViewportUnitQuirk const):
(WebCore::Quirks::needsCustomUserAgentData const):
(WebCore::Quirks::needsNavigatorUserAgentDataQuirk const):
(WebCore::Quirks::needsNowPlayingFullscreenSwapQuirk const):
(WebCore::Quirks::needsSuppressPostLayoutBoundaryEventsQuirk const):
(WebCore::Quirks::needsTikTokOverflowingContentQuirk const):
(WebCore::Quirks::needsWebKitMediaTextTrackDisplayQuirk const):
(WebCore::Quirks::shouldSupportHoverMediaQueries const):
(WebCore::Quirks::shouldRewriteMediaRangeRequestForURL const):
(WebCore::Quirks::shouldEnterNativeFullscreenWhenCallingElementRequestFullscreenQuirk const):
(WebCore::Quirks::shouldDelayReloadWhenRegisteringServiceWorker const):
(WebCore::Quirks::shouldDisableDOMAudioSessionQuirk const):
(WebCore::handle365ScoresQuirks):
(WebCore::handleNYTimesQuirks):
(WebCore::handleASQuirks):
(WebCore::handleATTQuirks):
(WebCore::handleCBSSportsQuirks):
(WebCore::handleSteamQuirks):
(WebCore::handleCNNQuirks):
(WebCore::handleDigitalTrendsQuirks):
(WebCore::handleGizmodoQuirks):
(WebCore::handleInstagramQuirks):
(WebCore::handleMailChimpQuirks):
(WebCore::handleRalphLaurenQuirks):
(WebCore::handleSlackQuirks):
(WebCore::handleWalmartQuirks):
(WebCore::handleScriptToEvaluateBeforeRunningScriptFromURLQuirk):
(WebCore::handleICloudQuirks):
(WebCore::handleTMobileQuirks):
(WebCore::handleCEACStateGovQuirks):
(WebCore::handleTrixEditorQuirks):
(WebCore::handleWeatherQuirks):
(WebCore::handleTikTokQuirks):
(WebCore::handleDisneyPlusQuirks):
(WebCore::handleGuardianQuirks):
(WebCore::handleMaxQuirks):
(WebCore::handleBaiduQuirks):
(WebCore::handleCodepenQuirks):
(WebCore::handleWarbyParkerQuirks):
(WebCore::handleACTestingQuirks):
(WebCore::handleDailyMailCoUkQuirks):
(WebCore::handleClaudeQuirks):
(WebCore::handleYCombinatorQuirks):
(WebCore::handleSoylentQuirks):
(WebCore::handleFacebookQuirks):
(WebCore::handleFacebookMessengerQuirks):
(WebCore::handleForbesQuirks):
(WebCore::handleRedditQuirks):
(WebCore::handleAmazonQuirks):
(WebCore::handleBBCQuirks):
(WebCore::handleBankOfAmericaQuirks):
(WebCore::handleBingQuirks):
(WebCore::handleBungalowQuirks):
(WebCore::handleDescriptQuirks):
(WebCore::handleESPNQuirks):
(WebCore::handleGoogleQuirks):
(WebCore::handleHBOMaxQuirks):
(WebCore::handleHotelsQuirks):
(WebCore::handleHuluQuirks):
(WebCore::handleIMDBQuirks):
(WebCore::handleLiveQuirks):
(WebCore::handleMarcusQuirks):
(WebCore::handleMediumQuirks):
(WebCore::handleMenloSecurityQuirks):
(WebCore::handleNBAQuirks):
(WebCore::handleNHLQuirks):
(WebCore::handleNetflixQuirks):
(WebCore::handlePandoraQuirks):
(WebCore::handlePremierLeagueQuirks):
(WebCore::handleSFUSDQuirks):
(WebCore::handleSharePointQuirks):
(WebCore::handleSoundCloudQuirks):
(WebCore::handleSpotifyQuirks):
(WebCore::handleVictoriasSecretQuirks):
(WebCore::handleTympanusQuirks):
(WebCore::handleVimeoQuirks):
(WebCore::handleWeeblyQuirks):
(WebCore::handleWikipediaQuirks):
(WebCore::handleTwitterXQuirks):
(WebCore::handleYouTubeQuirks):
(WebCore::handleZillowQuirks):
(WebCore::handleZomatoQuirks):
(WebCore::handleZoomQuirks):
(WebCore::handleCapitalGroupQuirks):
(WebCore::handleCrunchyRollQuirks):
(WebCore::Quirks::determineRelevantQuirks):
(WebCore::Quirks::hasRelevantQuirks const):
* Source/WebCore/page/Quirks.h:
* Source/WebCore/page/QuirksData.h:
* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::Adjuster::adjustForSiteSpecificQuirks const):

Canonical link: <a href="https://commits.webkit.org/303068@main">https://commits.webkit.org/303068@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/08e242b347c8246309226eee7a97c6bb7658a052

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/131100 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/3427 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/42061 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/138541 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/82794 "Built successfully") | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/98521089-0d8b-4748-a8c7-1f99956ad64b) 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/132970 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/3419 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/3268 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/99873 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/67640 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/302b0911-e69d-46e1-9139-64ab756e86e3) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/134046 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/2459 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/117359 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/80579 "Passed tests") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/0d60450d-eb18-44ec-97f3-bdb2fe356462) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/2381 "Build was cancelled. Recent messages:OS: Tahoe (26.0), Xcode: 26.0.1") | [![loading-orange](https://github-production-user-asset-6210df.s3.amazonaws.com/3098702/291015173-08c448be-ac0a-4fd6-92a3-8165057445b7.png) 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/46 "Build is in progress. Recent messages:OS: Tahoe (26.0), Xcode: 26.0.1; Skipping applying patch since patch_id isn't provided; Checked out pull request; 3 api tests failed or timed out; re-run-api-tests (cancelled)") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/81788 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/110965 "Passed tests") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/35527 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/141035 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/3170 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/35995 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/108391 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/3218 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/2831 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/108345 "Passed tests") | | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/2394 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/113689 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/56228 "Built successfully") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/20408 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/3237 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/32111 "Passed tests") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/3057 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/66633 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/3258 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/3167 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

stwrt commented on Source/WebCore/page/Quirks.cpp:
```diff
@@ -2160,1043 +2436,988 @@ void Quirks::setTopDocumentURLForTesting(URL&& url)
+    if (quirksDomainString != "scribd.com"_s) [[unlikely]]
+        return;
+
+    // Remove this once rdar://142573562 is resolved.
```
Please verify this is the correct radar.

The comments in the quirks show it is also being used for steam too.

brentfulgham replied:
Good catch! This was an old copy/paste error. I will fix it. Reviewing that also revealed that an old `std::optional<bool>` was present we don't need anymore.

stwrt commented on Source/WebCore/page/Quirks.cpp:
```diff
@@ -2160,1043 +2436,988 @@ void Quirks::setTopDocumentURLForTesting(URL&& url)
+static void handleMaxQuirks(QuirksData& quirksData, const URL& /* quirksURL */, const String& quirksDomainString, const URL&  /* documentURL */)
 {
-    if (quirksDomainString != "max.com"_s)
+    if (quirksDomainString != "max.com"_s) [[unlikely]]
```
We use quite a few checks of `[[unlikely]``. Shouldn't these be ``likely``, since there is a good chance it is not the domain?

brentfulgham replied:
No -- we only enter these handlers if the domain matches "e.g. 'max'" in this case. So if a website was 'max.net' or 'max.edu' we would want to bail out here, But that's unlikely since most users (especially those running in our test system to generate PGO) will only be visiting the real site.

brentfulgham committed:
Use a BitSet to hold quirk state
https://bugs.webkit.org/show_bug.cgi?id=302333
rdar://164490282

Reviewed by Brandon Stewart.

Rather than an ever-growing set of `bool` values, let's represent quirks state using a
BitSet. This allows us to do a quick check of whether *any* quirks are active in the
current domain, rather than individually testing each one. In theory, this should be a
small performance benefit since most sites do not have any relevant quirk and quickly
determining this requires fewer steps.

Tested by existing Quirks regression tests.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::needsFormControlToBeMouseFocusable const):
(WebCore::Quirks::needsAutoplayPlayPauseEvents const):
(WebCore::Quirks::needsSeekingSupportDisabled const):
(WebCore::Quirks::needsPerDocumentAutoplayBehavior const):
(WebCore::Quirks::shouldAutoplayWebAudioForArbitraryUserGesture const):
(WebCore::Quirks::hasBrokenEncryptedMediaAPISupportQuirk const):
(WebCore::Quirks::isTouchBarUpdateSuppressedForHiddenContentEditable const):
(WebCore::Quirks::isNeverRichlyEditableForTouchBar const):
(WebCore::Quirks::shouldSuppressAutocorrectionAndAutocapitalizationInHiddenEditableAreas const):
(WebCore::Quirks::shouldDispatchSyntheticMouseEventsWhenModifyingSelection const):
(WebCore::Quirks::needsYouTubeMouseOutQuirk const):
(WebCore::Quirks::shouldDisableWritingSuggestionsByDefault const):
(WebCore::Quirks::storageAccessUserAgentStringQuirkForDomain):
(WebCore::Quirks::shouldDisableElementFullscreenQuirk const):
(WebCore::Quirks::shouldDispatchSimulatedMouseEvents const):
(WebCore::Quirks::shouldDispatchedSimulatedMouseEventsAssumeDefaultPrevented const):
(WebCore::Quirks::shouldPreventDispatchOfTouchEvent const):
(WebCore::Quirks::shouldAvoidResizingWhenInputViewBoundsChange const):
(WebCore::Quirks::shouldDisablePointerEventsQuirk const):
(WebCore::Quirks::needsDeferKeyDownAndKeyPressTimersUntilNextEditingCommand const):
(WebCore::Quirks::needsGMailOverflowScrollQuirk const):
(WebCore::Quirks::needsYouTubeOverflowScrollQuirk const):
(WebCore::Quirks::needsPrimeVideoUserSelectNoneQuirk const):
(WebCore::Quirks::needsFacebookRemoveNotSupportedQuirk const):
(WebCore::Quirks::needsScrollbarWidthThinDisabledQuirk const):
(WebCore::Quirks::needsBodyScrollbarWidthNoneDisabledQuirk const):
(WebCore::Quirks::needsFullscreenDisplayNoneQuirk const):
(WebCore::Quirks::needsFullscreenObjectFitQuirk const):
(WebCore::Quirks::needsZomatoEmailLoginLabelQuirk const):
(WebCore::Quirks::needsGoogleMapsScrollingQuirk const):
(WebCore::Quirks::needsGoogleTranslateScrollingQuirk const):
(WebCore::Quirks::shouldSilenceResizeObservers const):
(WebCore::Quirks::shouldSilenceWindowResizeEventsDuringApplicationSnapshotting const):
(WebCore::Quirks::shouldSilenceMediaQueryListChangeEvents const):
(WebCore::Quirks::shouldAvoidScrollingWhenFocusedContentIsVisible const):
(WebCore::Quirks::shouldUseLegacySelectPopoverDismissalBehaviorInDataActivation const):
(WebCore::Quirks::shouldIgnoreAriaForFastPathContentObservationCheck const):
(WebCore::Quirks::shouldIgnoreViewportArgumentsToAvoidExcessiveZoom const):
(WebCore::Quirks::shouldIgnoreViewportArgumentsToAvoidEnlargedView const):
(WebCore::Quirks::shouldOpenAsAboutBlank const):
(WebCore::Quirks::needsPreloadAutoQuirk const):
(WebCore::Quirks::shouldBypassBackForwardCache const):
(WebCore::Quirks::shouldBypassAsyncScriptDeferring const):
(WebCore::Quirks::shouldEnableFacebookFlagQuirk const):
(WebCore::Quirks::applyFacebookFlagQuirk):
(WebCore::Quirks::shouldEnableLegacyGetUserMediaQuirk const):
(WebCore::Quirks::shouldDisableImageCaptureQuirk const):
(WebCore::Quirks::shouldEnableCameraAndMicrophonePermissionStateQuirk const):
(WebCore::Quirks::shouldEnableRemoteTrackLabelQuirk const):
(WebCore::Quirks::shouldEnableSpeakerSelectionPermissionsPolicyQuirk const):
(WebCore::Quirks::shouldEnableEnumerateDeviceQuirk const):
(WebCore::Quirks::shouldEnableRTCEncodedStreamsQuirk const):
(WebCore::Quirks::shouldUnloadHeavyFrame const):
(WebCore::Quirks::needsCanPlayAfterSeekedQuirk const):
(WebCore::Quirks::shouldLayOutAtMinimumWindowWidthWhenIgnoringScalingConstraints const):
(WebCore::Quirks::shouldNotAutoUpgradeToHTTPSNavigation):
(WebCore::Quirks::triggerOptionalStorageAccessQuirk const):
(WebCore::Quirks::needsVP9FullRangeFlagQuirk const):
(WebCore::Quirks::requiresUserGestureToPauseInPictureInPicture const):
(WebCore::Quirks::returnNullPictureInPictureElementDuringFullscreenChange const):
(WebCore::Quirks::requiresUserGestureToLoadInPictureInPicture const):
(WebCore::Quirks::blocksReturnToFullscreenFromPictureInPictureQuirk const):
(WebCore::Quirks::blocksEnteringStandardFullscreenFromPictureInPictureQuirk const):
(WebCore::Quirks::shouldDisableEndFullscreenEventWhenEnteringPictureInPictureFromFullscreenQuirk const):
(WebCore::Quirks::allowLayeredFullscreenVideos const):
(WebCore::Quirks::shouldDisableFullscreenVideoAspectRatioAdaptiveSizing const):
(WebCore::Quirks::shouldEnableFontLoadingAPIQuirk const):
(WebCore::Quirks::shouldDisableAdSkippingInPip const):
(WebCore::Quirks::needsVideoShouldMaintainAspectRatioQuirk const):
(WebCore::Quirks::shouldExposeShowModalDialog const):
(WebCore::Quirks::shouldNavigatorPluginsBeEmpty const):
(WebCore::Quirks::shouldDisableLazyIframeLoadingQuirk const):
(WebCore::Quirks::shouldDisableFetchMetadata const):
(WebCore::Quirks::shouldBlockFetchWithNewlineAndLessThan const):
(WebCore::Quirks::shouldDisablePushStateFilePathRestrictions const):
(WebCore::Quirks::needsConfigurableIndexedPropertiesQuirk const):
(WebCore::Quirks::advancedPrivacyProtectionSubstituteDataURLForScriptWithFeatures const):
(WebCore::Quirks::needsResettingTransitionCancelsRunningTransitionQuirk const):
(WebCore::Quirks::shouldDisableDataURLPaddingValidation const):
(WebCore::Quirks::needsDisableDOMPasteAccessQuirk const):
(WebCore::Quirks::shouldPreventOrientationMediaQueryFromEvaluatingToLandscape const):
(WebCore::Quirks::shouldFlipScreenDimensions const):
(WebCore::Quirks::requirePageVisibilityToPlayAudioQuirk const):
(WebCore::Quirks::shouldIgnorePlaysInlineRequirementQuirk const):
(WebCore::Quirks::shouldUseEphemeralPartitionedStorageForDOMCookies const):
(WebCore::Quirks::needsLaxSameSiteCookieQuirk const):
(WebCore::Quirks::shouldIgnoreTextAutoSizing const):
(WebCore::Quirks::scriptToEvaluateBeforeRunningScriptFromURL):
(WebCore::Quirks::shouldHideCoarsePointerCharacteristics const):
(WebCore::Quirks::implicitMuteWhenVolumeSetToZero const):
(WebCore::Quirks::shouldDispatchPointerOutAfterHandlingSyntheticClick const):
(WebCore::Quirks::needsZeroMaxTouchPointsQuirk const):
(WebCore::Quirks::needsChromeMediaControlsPseudoElement const):
(WebCore::Quirks::shouldHideSoftTopScrollEdgeEffectDuringFocus const):
(WebCore::Quirks::shouldTreatAddingMouseOutEventListenerAsContentChange const):
(WebCore::Quirks::shouldSynthesizeTouchEventsAfterNonSyntheticClick const):
(WebCore::Quirks::shouldIgnoreContentObservationForClick const):
(WebCore::Quirks::needsMozillaFileTypeForDataTransfer const):
(WebCore::Quirks::needsBingGestureEventQuirk const):
(WebCore::Quirks::shouldAvoidStartingSelectionOnMouseDownOverPointerCursor const):
(WebCore::Quirks::shouldReuseLiveRangeForSelectionUpdate const):
(WebCore::Quirks::needsPointerTouchCompatibility const):
(WebCore::Quirks::needsFacebookStoriesCreationFormQuirk const):
(WebCore::Quirks::needsHotelsAnimationQuirk const):
(WebCore::Quirks::needsClaudeSidebarViewportUnitQuirk const):
(WebCore::Quirks::needsCustomUserAgentData const):
(WebCore::Quirks::needsNavigatorUserAgentDataQuirk const):
(WebCore::Quirks::needsNowPlayingFullscreenSwapQuirk const):
(WebCore::Quirks::needsSuppressPostLayoutBoundaryEventsQuirk const):
(WebCore::Quirks::needsTikTokOverflowingContentQuirk const):
(WebCore::Quirks::needsWebKitMediaTextTrackDisplayQuirk const):
(WebCore::Quirks::shouldSupportHoverMediaQueries const):
(WebCore::Quirks::shouldRewriteMediaRangeRequestForURL const):
(WebCore::Quirks::shouldEnterNativeFullscreenWhenCallingElementRequestFullscreenQuirk const):
(WebCore::Quirks::shouldDelayReloadWhenRegisteringServiceWorker const):
(WebCore::Quirks::shouldDisableDOMAudioSessionQuirk const):
(WebCore::handle365ScoresQuirks):
(WebCore::handleNYTimesQuirks):
(WebCore::handleASQuirks):
(WebCore::handleATTQuirks):
(WebCore::handleCBSSportsQuirks):
(WebCore::handleSteamQuirks):
(WebCore::handleCNNQuirks):
(WebCore::handleDigitalTrendsQuirks):
(WebCore::handleGizmodoQuirks):
(WebCore::handleInstagramQuirks):
(WebCore::handleMailChimpQuirks):
(WebCore::handleRalphLaurenQuirks):
(WebCore::handleSlackQuirks):
(WebCore::handleWalmartQuirks):
(WebCore::handleScriptToEvaluateBeforeRunningScriptFromURLQuirk):
(WebCore::handleICloudQuirks):
(WebCore::handleTMobileQuirks):
(WebCore::handleCEACStateGovQuirks):
(WebCore::handleTrixEditorQuirks):
(WebCore::handleWeatherQuirks):
(WebCore::handleTikTokQuirks):
(WebCore::handleDisneyPlusQuirks):
(WebCore::handleGuardianQuirks):
(WebCore::handleMaxQuirks):
(WebCore::handleBaiduQuirks):
(WebCore::handleCodepenQuirks):
(WebCore::handleWarbyParkerQuirks):
(WebCore::handleACTestingQuirks):
(WebCore::handleDailyMailCoUkQuirks):
(WebCore::handleClaudeQuirks):
(WebCore::handleYCombinatorQuirks):
(WebCore::handleSoylentQuirks):
(WebCore::handleFacebookQuirks):
(WebCore::handleFacebookMessengerQuirks):
(WebCore::handleForbesQuirks):
(WebCore::handleRedditQuirks):
(WebCore::handleAmazonQuirks):
(WebCore::handleBBCQuirks):
(WebCore::handleBankOfAmericaQuirks):
(WebCore::handleBingQuirks):
(WebCore::handleBungalowQuirks):
(WebCore::handleDescriptQuirks):
(WebCore::handleESPNQuirks):
(WebCore::handleGoogleQuirks):
(WebCore::handleHBOMaxQuirks):
(WebCore::handleHotelsQuirks):
(WebCore::handleHuluQuirks):
(WebCore::handleIMDBQuirks):
(WebCore::handleLiveQuirks):
(WebCore::handleMarcusQuirks):
(WebCore::handleMediumQuirks):
(WebCore::handleMenloSecurityQuirks):
(WebCore::handleNBAQuirks):
(WebCore::handleNHLQuirks):
(WebCore::handleNetflixQuirks):
(WebCore::handlePandoraQuirks):
(WebCore::handlePremierLeagueQuirks):
(WebCore::handleSFUSDQuirks):
(WebCore::handleSharePointQuirks):
(WebCore::handleSoundCloudQuirks):
(WebCore::handleSpotifyQuirks):
(WebCore::handleVictoriasSecretQuirks):
(WebCore::handleTympanusQuirks):
(WebCore::handleVimeoQuirks):
(WebCore::handleWeeblyQuirks):
(WebCore::handleWikipediaQuirks):
(WebCore::handleTwitterXQuirks):
(WebCore::handleYouTubeQuirks):
(WebCore::handleZillowQuirks):
(WebCore::handleZomatoQuirks):
(WebCore::handleZoomQuirks):
(WebCore::handleCapitalGroupQuirks):
(WebCore::handleCrunchyRollQuirks):
(WebCore::Quirks::determineRelevantQuirks):
(WebCore::Quirks::hasRelevantQuirks const):
* Source/WebCore/page/Quirks.h:
* Source/WebCore/page/QuirksData.h:
* Source/WebCore/style/StyleAdjuster.cpp:
(WebCore::Style::Adjuster::adjustForSiteSpecificQuirks const):

Canonical link: https://commits.webkit.org/303068@main
